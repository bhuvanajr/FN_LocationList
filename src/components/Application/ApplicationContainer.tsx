import React from 'react';
import { ContainerFooter } from './ContainerFooter';
import { ContainerHeader } from './ContainerHeader';

export const ApplicationContainer: React.FC = props => {
  return (
    <div className="application-container">
      <header>
        <ContainerHeader />
      </header>
      <main className="container mx-auto">{props.children}</main>
      <footer>
        <ContainerFooter />
      </footer>
    </div>
  );
};
