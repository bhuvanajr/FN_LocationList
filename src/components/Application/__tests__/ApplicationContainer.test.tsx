import { render } from '@testing-library/react';
import React from 'react';
import { ApplicationContainer } from '../ApplicationContainer';

jest.mock('../ContainerFooter.tsx', () => ({
  ContainerFooter: () => <div>ContainerFooter</div>,
}));

jest.mock('../ContainerHeader.tsx', () => ({
  ContainerHeader: () => <div>ContainerHeader</div>,
}));

test('should render ApplicationContainer correctly', () => {
  const { container } = render(<ApplicationContainer>Test message</ApplicationContainer>);
  expect(container).toMatchSnapshot();
});
