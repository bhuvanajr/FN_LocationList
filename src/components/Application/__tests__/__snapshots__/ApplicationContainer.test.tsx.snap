// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ApplicationContainer correctly 1`] = `
<div>
  <div
    class="application-container"
  >
    <header>
      <div>
        ContainerHeader
      </div>
    </header>
    <main
      class="container mx-auto"
    >
      Test message
    </main>
    <footer>
      <div>
        ContainerFooter
      </div>
    </footer>
  </div>
</div>
`;
